code/modules/spells/targeted/absorb.dm

```diff
@@ -0,0 +1,61 @@
+/spell/targeted/absorb
+	name = "Absorb"
+	desc = "This spell allows you to absorb the magical abilities from fallen wizards. You must be standing over their lifeless or mutilated body in order for the spell to work."
+	abbreviation = "DG"
+	user_type = USER_TYPE_SPELLBOOK
+
+	school = "evocation"
+	charge_max = 100
+	spell_flags = IS_HARMFUL
+	invocation_type = SpI_SHOUT
+	range = 0
+	cooldown_min = 200
+
+	hud_state = "wiz_disint_old"
+
+
+/spell/targeted/absorb/invocation()
+	invocation = pick("B'TES TH' DUST", "H'STA LA V'STA", "S'AY'NARA", "G'ME 'VER")
+	..()
+
+/spell/targeted/absorb/cast(var/list/targets)
+	..()
+	var/mob/living/L = holder
+	for(var/mob/living/target in targets)
+		if(ishuman(target) || ismonkey(target))
+			var/mob/living/carbon/C = target
+			if(!C.mind)
+				to_chat(holder, "<span class='warning'>This being is devoid of any magic.</span>")
+			else if(!C.isDead() && !C.isInCrit())
+				to_chat(holder, "<span class='warning'>Your enemy is still full of life! Kill or maim them!</span>")
+			else if(!iswizard(target))	//Just in case the wizard manages to summon spells for the crew. No fun allowed.
+				to_chat(holder, "<span class='warning'>You can only absorb spells from other Wizards.</span>")
+			else
+				var/obj/effect/cult_ritual/feet_portal/P = new (holder.loc, holder, src)
+				if(do_after(holder, target, 5 SECONDS))
+					qdel(P)
+					var/hasAbsorbed = FALSE
+					for(var/spell/targetspell in C.spell_list)
+						if(!is_type_in_list(targetspell, L.spell_list))
```

Comment:
`is_type_in_list` will modify `L.spell_list` which might lead to Unforeseen Consequences, I suggest implementing the for loop manually in this case.